src/audiences/edit/index.js

```diff
@@ -0,0 +1,235 @@
+import React, { useState, useEffect, Fragment } from 'react';
+import ReactDOM from 'react-dom';
+import Editor from './components/editor';
+import Group from './components/group';
+import Options from './components/options';
+import Rule from './components/rule';
+import SelectInclude from './components/select-include';
+import {
+	defaultAudience,
+	defaultGroup,
+	defaultRule,
+} from './data/defaults';
+import {
+	getFields,
+} from './data';
+
+const { __ } = wp.i18n;
+const { Button } = wp.components;
+
+// Check for standard post edit page options meta box.
+const AudienceOptionsUI = document.getElementById( 'altis-analytics-audience-options' );
+
+const Edit = props => {
+	// Set the audience from props or use default.
+	const [ audience, setAudience ] = useState( props.audience || defaultAudience );
+	// Set the dropdown fields from props or fetch.
+	const [ fields, setFields ] = useState( props.fields || [] );
+	// Show errors.
+	const [ error, setError ] = useState( null );
+
+	// Fetch fields data.
+	useEffect( () => {
+		if ( fields.length ) {
+			return;
+		}
+		( async () => {
+			const fieldsResponse = await getFields();
+			if ( ! fieldsResponse instanceof Array && fieldsResponse.error ) {
+				setError( fieldsResponse.error );
+			} else {
+				if ( error ) {
+					setError( null );
+				}
+				setFields( fieldsResponse );
+			}
+		} )();
+	}, [ fields, error ] );
+
+	// Audience update helpers.
+	const updateAudience = ( newAudience ) => {
+		setAudience( audienceToUpdate => {
+			return { ...audienceToUpdate, ...newAudience };
+		} );
+	};
+
+	const updateGroup = ( groupId, group ) => {
+		const groups = audience.groups.slice();
+		const newGroup = Object.assign( {}, groups[ groupId ], group );
+		groups.splice( groupId, 1, newGroup );
+		updateAudience( { groups } );
+	};
+
+	const updateRule = ( groupId, ruleId, rule ) => {
+		const rules = audience.groups[ groupId ].rules.slice();
+		const newRule = Object.assign( {}, rules[ ruleId ], rule );
+		rules.splice( ruleId, 1, newRule );
+		updateGroup( groupId, { rules } );
+	};
+
+	return (
+		<Editor>
+			{ error && (
+				<div className="error msg">
+					<p>{ error.message }</p>
+				</div>
+			) }
+
+			{ AudienceOptionsUI && ReactDOM.createPortal( <Options audience={ audience } />, AudienceOptionsUI ) }
+
+			<div className="audience-editor__include">
+				<SelectInclude
+					onChange={ e => updateAudience( { include: e.target.value } ) }
+					value={ audience.include }
+					name="audience[include]"
+					label={ __( 'groups', 'altis-analytics' ) }
```

Comment:
https://engineering.hmn.md/standards/style/react/#align-and-sort-html-properties
```suggestion
					name="audience[include]"
					label={ __( 'groups', 'altis-analytics' ) }
					value={ audience.include }
					onChange={ e => updateAudience( { include: e.target.value } ) }
```